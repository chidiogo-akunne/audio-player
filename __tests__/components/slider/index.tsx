import Slider from '@components/slider';
import renderer from 'react-test-renderer';
import ThemeProvider from '@theme';
import { ContextThemeProvider } from '@context/index';

jest.useFakeTimers();

const onPressMock = jest.fn();

describe('Slider', () => {
  const tree = renderer.create(
    <ThemeProvider>
      <ContextThemeProvider>
        <Slider
          positionMillis={0}
          durationMillis={23456}
          onSlidingComplete={onPressMock}
        />
      </ContextThemeProvider>
    </ThemeProvider>
  );
  it(`renders correctly`, async () => {
    await expect(tree).toMatchSnapshot();
  });
});
