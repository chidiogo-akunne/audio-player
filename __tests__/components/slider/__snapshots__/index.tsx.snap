// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "marginTop": 59,
          "width": "100%",
        },
      ]
    }
    testID="slider"
  >
    <RNCSlider
      disabled={false}
      enabled={true}
      inverted={false}
      maximumTrackTintColor="#989898"
      maximumValue={1}
      minimumTrackTintColor="#686868"
      minimumValue={0}
      onChange={null}
      onRNCSliderSlidingComplete={[Function]}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={null}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={0}
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "height": 98,
            "width": "100%",
          },
        ]
      }
      tapToSeek={false}
      thumbImage={null}
      value={0}
    />
  </View>
</View>
`;
